### PR TITLE
[MOB-222] Update init to 3.0.0

### PR DESCRIPTION
**Pull request by tothszabi**
### Checklist
- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version
Requires a minor release.

### Changes
This PR updates the `bitrise-init` dependency and removes all Xamarin references.

### CODE REVIEW

**Review by ofalvai — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Project Scanner
 
-Scans repository for iOS, macOS, Android, Xamarin, Fastlane and Cordova projects.
```
Comment (godrei) [inline README.md:3]: NOTE: ionic and react native is missing from the list

Comment (godrei) [reply README.md:3]: NOTE: README could be auto-generated using the [generate_readme wf](https://github.com/bitrise-steplib/steps-xcode-archive/blob/master/bitrise.yml#L45).

File: step.yml [hunk 2]
```diff
@@ -1,15 +1,12 @@
 title: "Project scanner"
-summary: Scans repository for iOS, macOS, Android, Xamarin, Fastlane and Cordova projects
+summary: Scans repository for iOS, macOS, Android, Fastlane and Cordova projects
```
Comment (godrei) [inline step.yml:2]: NOTE: some supported project types are missing here

Comment (tothszabi) [reply step.yml]: I have added all of the missing project types.

File: README.md (hunk 1 above)
Comment (tothszabi) [reply README.md:3]: I have added the readme generation workflow and also generated a new readme.

**Review by ofalvai — APPROVED** [review]

**Pull request merged**